In [25]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
import plotly.colors as colors
pio.templates.default = "plotly_white"

In [26]:
data=pd.read_csv('Sample_Superstore.csv',encoding='latin-1')

In [27]:
data.head(2)

Row ID        Order ID Order Date   Ship Date     Ship Mode Customer ID  \
0       1  CA-2016-152156  11/8/2016  11/11/2016  Second Class    CG-12520   
1       2  CA-2016-152156  11/8/2016  11/11/2016  Second Class    CG-12520   

  Customer Name   Segment        Country       City  ... Postal Code  Region  \
0   Claire Gute  Consumer  United States  Henderson  ...       42420   South   
1   Claire Gute  Consumer  United States  Henderson  ...       42420   South   

        Product ID   Category Sub-Category  \
0  FUR-BO-10001798  Furniture    Bookcases   
1  FUR-CH-10000454  Furniture       Chairs   

                                        Product Name   Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.96         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.94         3   

   Discount    Profit  
0       0.0   41.9136  
1       0.0  219.5820  

[2 rows x 21 columns]

In [28]:
#Chacking Columns
data.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [29]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [30]:
# Chack Null Values
data.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [31]:
data.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

In [32]:
data[['Order Date']].head()

Order Date
0   11/8/2016
1   11/8/2016
2   6/12/2016
3  10/11/2015
4  10/11/2015

In [33]:
#Split Column and new cloemn add(Day, Month, Year)
data['Order_Day']=data['Order Date'].str.split('/').str[0]
data['Order_Month']=data['Order Date'].str.split('/').str[1]
data['Order_Year']=data['Order Date'].str.split('/').str[2]

In [34]:
data[['Order Date','Order_Day','Order_Month','Order_Year']].head()

Order Date Order_Day Order_Month Order_Year
0   11/8/2016        11           8       2016
1   11/8/2016        11           8       2016
2   6/12/2016         6          12       2016
3  10/11/2015        10          11       2015
4  10/11/2015        10          11       2015

In [35]:
#Change Datatype
data['Order_Day']=data['Order_Day'].astype(int)
data['Order_Month']=data['Order_Month'].astype(int)
data['Order_Year']=data['Order_Year'].astype(int)

In [36]:
data[['Order Date','Order_Day','Order_Month','Order_Year']].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Order Date   9994 non-null   object
 1   Order_Day    9994 non-null   int64 
 2   Order_Month  9994 non-null   int64 
 3   Order_Year   9994 non-null   int64 
dtypes: int64(3), object(1)
memory usage: 312.4+ KB


In [37]:
#1)You need to calculate the monthly sales of the store and identify which month had the highest sales and which 
# month had the lowest sales.
sales_by_month=data.groupby('Order_Month')['Sales'].sum().reset_index()

In [38]:
# Plot manually
Plot=px.line(
    sales_by_month,
    x='Order_Month',
    y='Sales',
    title='Month By Sales'
)
Plot.show()

#### Highest Sales Month: 17.0 - 112.1551K
#### Lowest Sales Month: 29.0 - 44.80669K

In [39]:
#2. You need to analyze sales based on product categories and determine which category has the
# lowest sales and which category has the highest sales.

sales_by_category=data.groupby('Category')['Sales'].sum().reset_index()

In [40]:
data['Category'].unique()

array(['Furniture', 'Office Supplies', 'Technology'], dtype=object)

In [41]:
chart=px.pie(
    sales_by_category,
    names='Category',
    values='Sales',
    hole=0.5,
    title='Catagory By Sales'
)
chart.show()

#### Catagory has Highest Sales : Office Supplies (36.4%)
#### Catagory has Lowest Sales : Technology (31.3%)

In [42]:
# 3. The sales analysis needs to be done based on sub-categories
sales_by_subcategory = data.groupby('Sub-Category')['Sales'].sum().reset_index()

bar = px.bar(sales_by_subcategory, 
             x='Sub-Category', 
             y='Sales',
             title='Sales Analysis by Sub-Category')
bar.show()

#### First Highest Sales Sub-Catagory : Phones (330.007k) and second highest sales :Chairs (328.4491k)
#### Last Highest Sales Sub-Catagory :Fasteners (3024.28)
 

In [43]:
# 4. You need to analyze the monthly profit from sales and determine which month had the highest profit.

profit_by_month = data.groupby('Order_Month')['Profit'].sum().reset_index()
fig = px.line(profit_by_month, 
              x='Order_Month', 
              y='Profit', 
              title='Monthly Profit Analysis',
            
              )
fig.show()

#### Highest Month Profit : month=23, profit=20.0966k

In [44]:
#5. Analyze the profit by category 
profit_by_category = data.groupby('Category')['Profit'].sum().reset_index()

fig = px.pie(profit_by_category, 
             values='Profit', 
             names='Category', 
             hole=0.5, 
             title='Category by Profit',          
             )
fig.show()

In [45]:
#5. Analyze the profit by subcategory 
sales_by_subcategory = data.groupby('Sub-Category')['Sales'].sum().reset_index()
fig = px.bar(sales_by_subcategory, 
             x='Sub-Category', 
             y='Sales', 
             title='Sales Analysis by Sub-Category'
             )
fig.show()

In [46]:
# 6. Analyze the sales and profit by customer segment
sales_profit_by_segment = data.groupby('Segment').agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()

fig = px.bar(
    sales_profit_by_segment,
    x='Segment',
    y=['Sales', 'Profit'],
    barmode='group',
    title='Sales and Profit by Customer Segment',
    labels={'value': 'Amount', 'Segment': 'Customer Segment'},
    text_auto=True,
    color_discrete_sequence=["#0EEEA7","#C37C54"]
)
fig.show()


In [47]:
# 7. Analyze the sales to profit ratio
sales_profit_by_segment = data.groupby('Segment').agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()
sales_profit_by_segment['Sales_to_Profit_Ratio'] = sales_profit_by_segment['Sales'] / sales_profit_by_segment['Profit']
print(sales_profit_by_segment[['Segment', 'Sales_to_Profit_Ratio']])

       Segment  Sales_to_Profit_Ratio
0     Consumer               8.659471
1    Corporate               7.677245
2  Home Office               7.125416


## Conclusion



#### Q1.) You need to calculate the monthly sales of the store and identify which month had the highest sales and which month had the lowest sales.
After calculating the monthly sales, it was observed that Month 17 recorded the highest sales with a total of ₹112.16K, while Month 29 had the lowest sales with a total of ₹44.81K.
#### Q2.)You need to analyze sales based on product categories and determine which category has the lowest sales and which category has the highest sales. 
Based on the analysis of sales by product category, Office Supplies accounted for the highest sales (36.4%), while Technology recorded the lowest sales (31.3%).
#### Q3.)The sales analysis needs to be done based on sub-categories.
The sales analysis by sub-category reveals that Phones generated the highest sales (₹330.01K), followed closely by Chairs (₹328.45K). In contrast, Fasteners recorded the lowest sales (₹3.02K).
#### Q4.) You need to analyze the monthly profit from sales and determine which month had the highest profit.
The monthly profit analysis shows that Month 23 achieved the highest profit, totaling ₹20.10K.
#### Q5.)Analyze the profit by category and sub-category.
The profit analysis by category and sub-category shows that Phones generated the highest profit, followed by Chairs. In contrast, Fasteners recorded the lowest profit among all sub-categories
#### Q6.) Analyze the sales and profit by customer segment.
From the analysis of sales and profit by customer segment, it is observed that the Consumer segment generated the highest sales (₹1.16M) and the highest profit (₹134.12K). The Corporate segment followed with moderate sales (₹706.15K) and profit (₹91.98K)**, while the Home Office segment recorded the lowest sales (₹429.65K) and lowest profit (₹60.30K).
#### Q7.)Analyze the sales to profit ratio.
The sales-to-profit ratio analysis reveals that the Consumer segment has the highest ratio (8.66), followed by the Corporate segment (7.68), and the Home Office segment (7.13).